app/process/watcher/consensus-message/consensus-message.go

```diff
@@ -107,10 +123,15 @@ func (ctw ConsensusTopicWatcher) subscribeToTopic(q *queue.Queue) {
 			*ctw.client.GetMirrorClient(),
 			func(response hedera.MirrorConsensusTopicResponse) {
 				log.Infof("Consensus Topic [%s] - Message incoming: [%s]", response.ConsensusTimestamp, ctw.topicID, response.Message)
-				// validate and check body
-				msg := &protomsg.TopicSignatureMessage{}
+
+				msg := &validatorproto.TopicSignatureMessage{}
+				err := proto.Unmarshal(response.Message, msg)
+				if err != nil {
+					log.Fatal(err)
+				}
+
 				publisher.Publish(msg, ctw.typeMessage, ctw.topicID, q)
-				err := ctw.statusRepository.UpdateLastFetchedTimestamp(ctw.topicID.String(), strconv.FormatInt(response.ConsensusTimestamp.Unix(), 10))
+				err = ctw.statusRepository.UpdateLastFetchedTimestamp(ctw.topicID.String(), strconv.FormatInt(response.ConsensusTimestamp.Unix(), 10))
 				if err != nil {
 					log.Fatal(err)
 				}
```

Comment:
This logic seems very similar to the one you are using when you are handling unprocessed messages. Maybe you can extract it to a function?